### PR TITLE
SER-4044 | Remove cache as part of RTBF

### PR DESCRIPTION
**Pull request by slayful**

### CODE REVIEW

**Review by pwojcik86 — APPROVED** [review]

File: includes/User.php [hunk 1]
```diff
@@ -2062,9 +2062,25 @@ public function deleteCache() {
...
 		$wgMemc->delete( self::getCacheKeyByName( $this->getName() ) ); // SUS-2945
+		$wgMemc->delete( self::getUserIdentityBoxCacheKey( $this->getId() ) );
+		$wgMemc->delete( self::getLookupContribsCoreCacheKey( $this->getId() ) );
 		$this->userPreferences()->deleteFromCache( $this->getId() );
 	}
 
+	/**
+	 * Returns memcached key for UserIdentityBox.
+	 */
+	public static function getUserIdentityBoxCacheKey( int $userId ): string {
```
Comment (lomiarz) [inline includes/User.php]: how about to make it non-static...just regular public function ? will that work ? and than you can easily call it like `$user->getUserIdentityBoxCacheKey()` without passing `userId`

File: includes/User.php [hunk 2]
```diff
@@ -2062,9 +2062,25 @@ public function deleteCache() {
...
+		return wfSharedMemcKey( 'user-identity-box-data0', $userId, /* version */ 3 );
+	}
+
+	/**
+	 * Returns memcached key for UserIdentityBox LookupContribsCore.
+	 */
+	public static function getLookupContribsCoreCacheKey( int $userId ): string {
+		return wfSharedMemcKey( 'LookupContribsCore', $userId );
```
Comment (lomiarz) [inline includes/User.php]: same as above here

File: extensions/wikia/UserProfilePageV3/UserIdentityBox.class.php [hunk 3]
```diff
@@ -266,9 +264,10 @@ private function getDefaultData( $data ) {
 	 * @return string
 	 */
 	private function getMemcUserIdentityDataKey() {
-		return wfSharedMemcKey( 'user-identity-box-data0', $this->user->getId(), self::CACHE_VERSION );
+		return User::getUserIdentityBoxCacheKey( $this->user->getId() );
```
Comment (lomiarz) [inline extensions/wikia/UserProfilePageV3/UserIdentityBox.class.php]: how about `$user->getUserIdentityBoxCacheKey()`

Comment (lomiarz) [reply extensions/wikia/UserProfilePageV3/UserIdentityBox.class.php]: ?

File: extensions/wikia/RecentChanges/RecentChangesFiltersStorage.class.php [hunk 4]
```diff
@@ -74,7 +74,7 @@ protected function setCache($values){
 	}
 
 	protected function getCache() {
-		return $this->memc->get($this->getCacheKey(), null);
+		return $this->memc->get( $this->getCacheKey() );
```
Comment (slayful) [inline extensions/wikia/RecentChanges/RecentChangesFiltersStorage.class.php:77]: `get` doesn't have a second argument.

File: extensions/wikia/UserProfilePageV3/UserIdentityBox.class.php [hunk 5]
```diff
@@ -1,15 +1,10 @@
 <?php
 
 class UserIdentityBox extends WikiaObject {
-	const CACHE_VERSION = 3;
 
-	/**
-	 * Prefixes to memc keys etc.
-	 */
```
Comment (slayful) [inline extensions/wikia/UserProfilePageV3/UserIdentityBox.class.php:8]: These weren't really that.

**Review by mateuszsip — APPROVED** [review]

**Pull request merged**